adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralBannerRenderer.m

```diff
@@ -0,0 +1,131 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterMintegralBannerRenderer.h"
+#import "GADMAdapterMintegralUtils.h"
+#import "GADMediationAdapterMintegralConstants.h"
+#include <stdatomic.h>
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKBanner/MTGBannerAdView.h>
+#import <MTGSDKBanner/MTGBannerAdViewDelegate.h>
+
+@interface GADMAdapterMintegralBannerRenderer ()
+<GADMediationBannerAd,
+MTGBannerAdViewDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralBannerRenderer{
+    /// Data used to render an banner ad.
+    GADMediationBannerAdConfiguration *_adConfiguration;
+    
+    // The completion handler to call when the ad loading succeeds or fails.
+    GADMediationBannerLoadCompletionHandler _adLoadCompletionHandler;
+    
+    // An ad event delegate to invoke when ad rendering events occur.
+    __weak id<GADMediationBannerAdEventDelegate> _adEventDelegate;
+    
+    // The Mintegral banner ad.
+    MTGBannerAdView *_bannerAdView;
+    
+}
+
+- (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
+                   completionHandler:
+(nonnull GADMediationBannerLoadCompletionHandler)completionHandler{
+    _adConfiguration = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationBannerLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationBannerAdEventDelegate>(
+                                                                      _Nullable id<GADMediationBannerAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationBannerAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+        
+    NSString *unitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    UIViewController *rootViewController = adConfiguration.topViewController;
+
+    if ([GADMAdapterMintegralUtils isEmpty:unitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    if ([GADMAdapterMintegralUtils isEmpty:adConfiguration.bidResponse]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"bid token cannot be nil.");
```

Comment:
Question, will Mintegral SDK return a failure callback if the bid response is empty? If it does, we probably don't need this check in the adapter.

Also, update to `Bid token ...` with first letter using upper case.

Reply:
ok,i will delete check code